examples/golang/envoy-go-filter-samples/pkg/http/api.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
```

Comment:
I ton of code that is in here as an example seems like core functionality? Shouldn't most of this be built into the extension and then the extension user just needs to provide the go code?

Reply:
> I ton of code that is in here as an example seems like core functionality? Shouldn't most of this be built into the extension and then the extension user just needs to provide the go code?

Yes, the extension users only need to provide the [plugin code](https://github.com/envoyproxy/envoy/tree/57ffc1cab6fb151f47f064e829bd924b0b9737a9/examples/golang/envoy-go-filter-samples/pkg/plugins/simple). This use case is mainly used to verify the correctness of API and the user's experience of writing go plugin.  So the current focus is to complete the design of the go extension API and merge it into the master, and then build a SIG to iterate on the go extension implementation together. Do you think it is okay?